src/lib/Libcmds/prepare_path.c

```diff
@@ -234,11 +253,14 @@ prepare_path(char *path_in, char *path_out)
 				ino = statbuf.st_ino;
 				if (stat(".", &statbuf) < 0) {
 					perror("prepare_path: cannot stat current directory:");
+					*path_out = '\0';
 					return (1);
 				}
 			}
 			if (dev == statbuf.st_dev && ino == statbuf.st_ino) {
-				strcpy(cwd, c);
+				size_t cwd_size = sizeof(cwd);
+				strncpy(cwd, c, cwd_size - 1);
```

Comment:
@bhroam: what I actually wanted originally was to save the entire 'sizeof(cwd) - 1' value into a variable, and then just use that in 2 places as in:
cwd_len = sizeof(cwd) - 1;
strncpy(cwd, c, cwd_len);
cwd[cwd_len] = '\0';

It's the repeating of reference (ex. sizeof(cwd) - 1) that I wanted to be improved, because if we change the code later, then less place to update.

Reply:
@bhroam's comment is inline with one of my earlier responses. The sizeof() calculations are done at compile time, so they do not affect the performance. Even subtracting one from the value is done at compile time, so it becomes a static integer. Let's figure out what we want here so I can make everyone happy. @bayucan?

Reply:
@mike0042 I can llive with what you have right now (current code).

Reply:
Converted it to snprintf() to be consistent.